Detectors/TRD/calibration/src/CalibratorNoise.cxx

```diff
@@ -14,17 +14,86 @@
+    int indexGlobal = HelperMethods::getGlobalChannelIndex(digit.getDetector(), digit.getROB(), digit.getMCM(), digit.getChannel());
+    auto& info = mChannelInfosDetailed[indexGlobal];
+    if (info.nEntries == 0) {
+      // the first time we see data from this channel we fill the detector information etc.
+      info.det = digit.getDetector();
+      info.sec = HelperMethods::getSector(digit.getDetector());
+      info.stack = HelperMethods::getStack(digit.getDetector());
+      info.layer = HelperMethods::getLayer(digit.getDetector());
+      info.row = digit.getPadRow();
+      info.col = digit.getPadCol();
+      info.isShared = digit.isSharedDigit();
+      info.channelGlb = HelperMethods::getChannelIndexInColumn(digit.getROB(), digit.getMCM(), digit.getChannel());
+      info.indexGlb = indexGlobal;
+    }
+    // the ADC information we always want to have
+    for (int i = 0; i < TIMEBINS; ++i) {
+      auto adc = digit.getADC()[i];
+      info.adcSum += adc;
+      info.adcSumSquared += adc * adc;
+      // mean and variance are calculated recursively
+      float meanCurrent = info.adcMean;
+      info.adcMean += (adc - meanCurrent) / (info.nEntries + 1);
+      info.variance += info.nEntries * (info.nEntries + 1) * (info.adcMean - meanCurrent) * (info.adcMean - meanCurrent);
+      info.nEntries += 1;
+    }
+  }
+  mNDigitsSeen += digits.size();
+  LOG(info) << "Processed " << digits.size() << " digits for this TF. Total number of processed digits: " << mNDigitsSeen;
+}
 
-Slot& CalibratorNoise::emplaceNewSlot(bool front, TFType tStart, TFType tEnd)
+void CalibratorNoise::collectChannelInfo()
 {
-  auto& container = getSlots();
-  auto& slot = front ? container.emplace_front(tStart, tEnd) : container.emplace_back(tStart, tEnd);
-  slot.setContainer(std::make_unique<PadAdcInfo>());
-  return slot;
+  for (const auto& channel : mChannelInfosDetailed) {
+    // create ChannelInfo object for array
+    if (channel.nEntries > 0) {
+      // we have seen data from this channel
+      bool isOK = true;
+      auto& ch = mCCDBObject.getChannel(channel.indexGlb);
+      ch.setMean(channel.adcMean);
+      ch.setRMS(std::sqrt(channel.variance / channel.nEntries));
+      uint8_t flags = 1 << ChannelInfo::Flag::IsSet;
+      if (ch.getMean() < mParams.meanAdcLow) {
+        flags |= (1 << ChannelInfo::Flag::LowMean);
+        isOK = false;
+      }
+      if (ch.getMean() > mParams.meanAdcHigh) {
+        flags |= ChannelInfo::Flag::HighMean;
+        isOK = false;
+      }
+      if (ch.getRMS() < mParams.rmsAdcLow && ch.getRMS() > mAlmostZero) {
+        flags |= (1 << ChannelInfo::Flag::LowRMS);
+        isOK = false;
+      }
+      if (ch.getRMS() > mParams.rmsAdcHigh) {
+        flags |= (1 << ChannelInfo::Flag::HighRMS);
+        isOK = false;
+      }
+      if (ch.getRMS() < mAlmostZero && std::abs(ch.getMean() - mParams.adcMasked) < mAlmostZero) {
+        flags |= (1 << ChannelInfo::Flag::IsMasked);
+        isOK = false;
+      }
```

Comment:
This classification is static, and I wonder what happens when we revisit the criteria for flagging channels?
Relating to Felix' comment about where and how we use the calibration, we might even want different criteria in different use cases.
The data for the classification is always there, so it could be done on the fly.

Reply:
True, currently it does not make sense to have it all. If we would want to go for a very slim object we could keep only the flags and store the settings like what is HighRMS etc (which are also configurable). In the CCDB object which keeps a vector with information for each channel. But since I don't see that we would use this object in the reconstruction but instead only to mask channels and compare to previous objects we shouldn't concentrate on reducing the size of the object